src/sim/kernel_workload.cc

```diff
@@ -104,12 +104,28 @@ KernelWorkload::initState()
     };
     if (params().object_file != "")  {
         if (params().addr_check) {
+            auto memory_size = system->memSize();
+            auto kernel_size = _end - _start;
             // Validate kernel mapping before loading binary
+            char error_msg_buffer [500];
```

Comment:
I don't see the reason for shooting ourselves in the foot with a fixed size array